src/core/models/answersconfig.js

```diff
@@ -0,0 +1,147 @@
+     * @type {string}
+     */
+    this.environment = rawConfig.environment;
+
+    /**
+     * Required, the key used for your Answers Experience
+     * @type {string}
+     */
+    this.experienceKey = rawConfig.experienceKey;
+
+    /**
+     *  Optional, initialize components here, invoked when the Answers component library is loaded/ready.
+     *  If components are not added here, they can also be added when the init promise resolves
+     * @type {function()}
+     */
+    this.onReady = rawConfig.onReady;
+
+    /**
+     * Optional*, Yext businessId, *required to send analytics events
+     * @type {number|string|*}
+     */
+    this.businessId = rawConfig.businessId;
+
+    /**
+     * Optional, if false, the library will not fetch pre-made templates. Only use change this to false if you provide a
+     *  template bundle in the `templateBundle` config option or implement custom renders for every component
+     * @type {boolean}
+     */
+    this.useTemplates = rawConfig.useTemplates;
+
+    /**
+     * Optional, additional templates to register with the renderer
+     * @type {Object}
+     */
+    this.templateBundle = rawConfig.templateBundle;
+
+    /**
+     * Optional, provide configuration for each vertical that is shared across components
+     * @type {VerticalPagesConfig}
+     */
+    this.verticalPages = new VerticalPagesConfig(rawConfig.verticalPages);
+
+    /**
+     * Optional, search specific settings
+     * @type {SearchConfig}
+     */
+    this.search = new SearchConfig(rawConfig.search);
+
+    /**
+     * Optional, vertical no results settings. See Vertical No Results
+     * @type {Object}
+     */
+    this.noResults = rawConfig.noResults;
+
+    /**
+     * Optional, the locale will affect how queries are interpreted and the results returned. Defaults to 'en'
+     * @type {string}
+     */
+    this.locale = rawConfig.locale;
+
+    /**
+     * Optional, the Answers Experience version to use for api requests
+     * @type {string}
+     */
+    this.experienceVersion = rawConfig.experienceVersion;
+
+    /**
+     * Optional, prints full Answers error details when set to \`true\`. Defaults to false.
+     * @type {boolean}
+     */
+    this.debug = rawConfig.debug;
+
+    /**
+     * Optional, If true, the search session is tracked. If false, there is no tracking. Defaults to true.
+     * @type {boolean}
+     */
+    this.sessionTrackingEnabled = rawConfig.sessionTrackingEnabled;
+
+    /**
+     * Optional, invoked when the state of any component changes
+     * @type {function()}
+     */
+    this.onStateChange = rawConfig.onStateChange;
+
+    /**
+     * Optional, analytics callback after a vertical search, see onVerticalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onVerticalSearch = rawConfig.onVerticalSearch;
+
+    /**
+     * Optional, analytics callback after a universal search, see onUniversalSearch Configuration for additional details
```

Comment:
does this "see onUniversalSearch" comment make sense here outside of the context of the README?